stable_baselines3/her/her_replay_buffer.py

```diff
@@ -304,14 +304,6 @@ def _sample_transitions(
 
         ep_lengths = self.episode_lengths[episode_indices]
 
-        # Special case when using the "future" goal sampling strategy
```

Comment:
why did you remove that?
this is different from the issue you raised, no?

Reply:
I think it's not.

Transitions are stored that way: 
```
(s0, g0, s1, g1), 
(s1, g1, s2, g2), 
(s2, g2, s3, g3), 
(s3, g3, s4, g4), 
```

The issue I raised is that the new desired goal are sampled from `achieved_goal` (first column) instead of `next_achieved_goal` (third column). Since you want _strictly next_ achieved goal, you add a +1 here

 https://github.com/DLR-RM/stable-baselines3/blob/44e53ff8115e8f4bff1d5218f10c8c7d1a4cfc12/stable_baselines3/her/her_replay_buffer.py#L255

If `s1` is sampled, you want the new desired goal to be either `s2`, `s3` or `s4`.
But, with the current implementation, you can only sample `s2` and `s3` as a new goal (that's the issue). 

Similarly, if `s3` is sampled, you want the new desired goal to be `s4` (only one choice).
But, with the current implementation, you cannot sample any goal. That's why this special case has been tackled.

With a correct implementation, even when the episode is a single transition `[(s0, g0, s1, g1)]`, you can sample a strictly future `achieved_goal`, by taking `s1`. So you don't need that fix anymore.

Reply:
I see, thanks for the clarification =)